src/runtime/pkg/resourcecontrol/utils_linux.go

```diff
@@ -21,9 +21,9 @@ import (
 // DefaultResourceControllerID runtime-determined location in the cgroups hierarchy.
 const DefaultResourceControllerID = "/vc"
 
-// ValidCgroupPathV1 returns a valid cgroup path for cgroup v1.
+// ValidCgroupPath returns a valid cgroup path.
 // see https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#cgroups-path
-func ValidCgroupPathV1(path string, systemdCgroup bool) (string, error) {
+func ValidCgroupPath(path string, systemdCgroup bool) (string, error) {
```

Comment:
The code of this function makes no sense because the second argument is `systemdCgroup := IsSystemdCgroup(path)`.

So the two first branches `if IsSystemdCgroup(path)` and `if systemdCgroup` perform the same check with different returns.

Please fix this somehow.

Reply:
Moved to #5153 to resolve the issue.